test/ca/mcgill/cs/stg/jetuml/graph/TestUsageScenariosClassDiagram.java

```diff
@@ -0,0 +1,694 @@
+		diagram.addEdge(aggrEdge, new Point2D.Double(70, 70), new Point2D.Double(170, 170));
+		// one point is invalid
+		diagram.addEdge(aggrEdge, new Point2D.Double(6, 7), new Point2D.Double(170, 170));
+		assertEquals(0, diagram.getEdges().size());
+		
+		diagram.addEdge(aggrEdge, new Point2D.Double(8, 10), new Point2D.Double(45, 48));
+		diagram.addEdge(assoEdge, new Point2D.Double(47, 49), new Point2D.Double(9, 17));
+		diagram.addEdge(depeEdge, new Point2D.Double(90, 93), new Point2D.Double(44, 49));
+		assertEquals(3, diagram.getEdges().size());
+		
+		diagram.addEdge(new AssociationEdge(), new Point2D.Double(47, 49), new Point2D.Double(50, 49));
+		assertEquals(4, diagram.getEdges().size());
+		
+		// not every edge is a valid to be self-edge
+		diagram.addEdge(geneEdge, new Point2D.Double(47, 49), new Point2D.Double(50, 49));
+		assertEquals(4, diagram.getEdges().size());
+	}
+	
+	@Test
+	public void testNoteEdgeCreation()
+	{
+		diagram.addNode(aClassNode, new Point2D.Double(5, 5));
+		diagram.addNode(aInterfaceNode, new Point2D.Double(44, 44));
+		diagram.addNode(aPackageNode, new Point2D.Double(87, 87));
+		diagram.addNode(aNoteNode, new Point2D.Double(134, 132));
+		
+		NoteEdge noteEdge1 = new NoteEdge();
+		NoteEdge noteEdge2 = new NoteEdge();
+		
+		// if begin with a non-NoteNode type, both point needs to be valid
+		diagram.addEdge(noteEdge1, new Point2D.Double(9, 9), new Point2D.Double(209,162));
+		assertEquals(0, diagram.getEdges().size());
+		diagram.addEdge(noteEdge1, new Point2D.Double(9, 9), new Point2D.Double(139,142));
+		assertEquals(1, diagram.getEdges().size());
+		assertEquals(noteEdge1.getStart(), aClassNode);
+		assertEquals(noteEdge1.getEnd(), aNoteNode);
+		
+		// if begin with a NoteNode, the end point can be anywhere
+		diagram.addEdge(noteEdge2, new Point2D.Double(138, 140), new Point2D.Double(9,9));
+		assertEquals(noteEdge2.getStart(), aNoteNode);
+		assertEquals(noteEdge2.getEnd().getClass(), new PointNode().getClass());
+		assertEquals(2, diagram.getEdges().size());
+
+	}
+	
+	/**
+	 * Testing Single Node Movement
+	 */
+	@Test
+	public void testSignelNodeMovement()
+	{
+		
+		diagram.addNode(aClassNode, new Point2D.Double(5, 5));
+		diagram.addNode(aInterfaceNode, new Point2D.Double(44, 44));
+		diagram.addNode(aPackageNode, new Point2D.Double(87, 87));
+		diagram.addNode(aNoteNode, new Point2D.Double(134, 132));
+		
+		aClassNode.translate(5, 5);
+		aInterfaceNode.translate(11, 19);
+		aPackageNode.translate(32, -42);
+		aNoteNode.translate(-5, 19);
+		
+		assertTrue(10 == aClassNode.getBounds().getX());
+		assertTrue(10 == aClassNode.getBounds().getY());
+		assertTrue(55 == aInterfaceNode.getBounds().getX());
+		assertTrue(63 == aInterfaceNode.getBounds().getY());
+		assertTrue((87+32) == aPackageNode.getBounds().getX());
+		assertTrue((87-42) == aPackageNode.getBounds().getY());
+		assertTrue(129 == aNoteNode.getBounds().getX());
+		assertTrue(151 == aNoteNode.getBounds().getY());
+	}
+	
+	/**
```

Comment:
Some typos. Here's the fixed version.
/**
+	 * Testing moving a selection of nodes and edges,
+	 * edges will be redrawn automatically.
+	 */